### PR TITLE
Issue 20 mapboundaries

### PR DESCRIPTION
**Pull request by MyK00L**

### CODE REVIEW

File: src/main/java/eu/eutampieri/catacombs/model/map/TileMapFactoryImpl.java [hunk 1]
```diff
@@ -9,13 +9,13 @@
...
-	private Random rand=new Random();
-	
+	private Random rand = new Random();
+
 	/**
 	 * A useful helper class for map generation
 	 */
 	private class Point {
-	  @Override
+		@Override
```
Comment (eutampieri) [inline src/main/java/eu/eutampieri/catacombs/model/map/TileMapFactoryImpl.java:18]: Use spaces to indent

File: src/main/java/eu/eutampieri/catacombs/model/map/TileMapFactoryImpl.java [hunk 2]
```diff
@@ -41,145 +42,172 @@ public boolean equals(Object obj) {
...
+
+		final int x;
+		final int y;
+
+		Point(int x, int y) {
+			this.x = x;
+			this.y = y;
+		}
+
+		int dist(Point o) {
```
Comment (eutampieri) [inline src/main/java/eu/eutampieri/catacombs/model/map/TileMapFactoryImpl.java:54]: We're in javaland, you could have used Point other

File: src/main/java/eu/eutampieri/catacombs/model/map/TileMapFactoryImpl.java [hunk 3]
```diff
@@ -41,145 +42,172 @@ public boolean equals(Object obj) {
...
-		int x=a.x;
-		while(y!=b.y && x!=b.x) {
-			if(rand.nextBoolean()) {
-				if(y<b.y)y++;
-				else y--;
+		int y = a.y;
+		int x = a.x;
+		while (y != b.y && x != b.x) {
+			if (rand.nextBoolean()) {
+				if (y < b.y)
```
Comment (eutampieri) [inline src/main/java/eu/eutampieri/catacombs/model/map/TileMapFactoryImpl.java:73]: Curly braces

File: src/main/java/eu/eutampieri/catacombs/model/map/TileMapFactoryImpl.java [hunk 4]
```diff
@@ -41,145 +42,172 @@ public boolean equals(Object obj) {
...
 				}
 			}
 		}
-		int minY=0,minX=0,maxY=0,maxX=0; // get map boundaries to shift coordinates and calculate map size
-		for(Point p:centers) {
-			if(p.y<minY)minY=p.y;
-			if(p.x<minX)minX=p.x;
-			if(p.y>maxY)maxY=p.y;
-			if(p.x>maxX)maxX=p.x;
+		int minY = 0, minX = 0, maxY = 0, maxX = 0; // get map boundaries to shift coordinates and calculate map size
```
Comment (eutampieri) [inline src/main/java/eu/eutampieri/catacombs/model/map/TileMapFactoryImpl.java:151]: No multiple variables declaration on one line

File: src/main/java/eu/eutampieri/catacombs/model/map/TileMapFactoryImpl.java [hunk 5]
```diff
@@ -41,145 +42,172 @@ public boolean equals(Object obj) {
...
 	 * @param minRoomDist minimum distance between two rooms' centers
-	 * @param maxRoomDist maximum distance with the closest room's center for each room center
-	 * @return A Tilemap with nRooms square rooms connected by corridors in a tree, plus some random corridors
-	 * minRoomDist > maxRoomSide is recommended
+	 * @param maxRoomDist maximum distance with the closest room's center for each
+	 *                    room center
+	 * @return A Tilemap with nRooms square rooms connected by corridors in a tree,
+	 *         plus some random corridors minRoomDist > maxRoomSide is recommended
 	 */
 	private TileMap normal(int nRooms, int minRoomSide, int maxRoomSide, int minRoomDist, int maxRoomDist) {
```
Comment (eutampieri) [inline src/main/java/eu/eutampieri/catacombs/model/map/TileMapFactoryImpl.java:111]: This method is huge

**Review by eutampieri — CHANGES_REQUESTED** [review]

File: src/main/java/eu/eutampieri/catacombs/model/map/TileMapFactoryImpl.java (hunk 2 above)
Comment (MyK00L) [reply src/main/java/eu/eutampieri/catacombs/model/map/TileMapFactoryImpl.java:54]: I could have also used thisIsTheCoordinateXThatRepresentsTheDistanceFromTheLeftBoundary, but I prefer not

File: src/main/java/eu/eutampieri/catacombs/model/map/TileMapFactoryImpl.java (hunk 5 above)
Comment (MyK00L) [reply src/main/java/eu/eutampieri/catacombs/model/map/TileMapFactoryImpl.java:111]: yes

**Pull request merged**